package/gluon-mesh-babel/luasrc/lib/gluon/upgrade/980-gluon-mesh-babel-firewall

```diff
@@ -0,0 +1,374 @@
+#!/usr/bin/lua
+
+-- TODO: replace REJECT by DROP. only use REJECT for debugging
+
+local uci = require('simple-uci').cursor()
+local site = require "gluon.site_config"
+
+-- TODO: this is too static and should be done non-persistently at boot-time
+-- return table with all interfaces that are running proto = gluon_mesh
+function find_meshes()
+	local nets={}
+
+	uci:foreach('network', 'interface',
+	function(s)
+		local section= s['.name']
+		if ( uci:get('network',section,'proto' ) == 'gluon_mesh' )
+		then
+			table.insert(nets, section )
+		end
+	end
+	)
+	return nets
+end
+
+-- find mesh firewall section
+function find_fw(zone)
+	local retval=""
+	uci:foreach('firewall','zone',
+	function(s)
+		local section=s['.name']
+		if ( uci:get('firewall',section,'name') == zone)
+		then
+			retval=section
+			return false
+		end
+	end
+	)
+	return retval
+end
+
+
+wansection=find_fw('wan')
+uci:delete('firewall', wansection)
+uci:delete('firewall', 'client')
+
+uci:section('firewall', 'zone', 'wan', {
+	name = 'wan',
+	network = { 'wan','wan6',},
+	input = 'REJECT',
+	output = 'REJECT',
+	forward = 'REJECT',
+	mtu_fix = '1',
+	conntrack = '1',
+	log = '1',
+}
+)
+uci:section('firewall', 'zone', 'client',
+{
+	name = 'client',
+	input = 'REJECT',
+	output = 'ACCEPT',
+	forward = 'REJECT',
+	network = {'client','local_node',},
+	conntrack = '1',
+	log = '1',
+}
+)
+uci:section('firewall', 'zone', 'mesh',
+{
+	name = 'mesh',
+	input = 'REJECT',
+	output = 'ACCEPT',
+	forward = 'REJECT',
+	conntrack = '1',
+	log = '1',
+}
+)
+uci:section('firewall', 'zone', 'l3roamd',
+{
+	name = 'l3roamd',
+	input = 'ACCEPT',
+	output = 'ACCEPT',
+	forward = 'REJECT',
+	device = 'l3roam+',
+	log = '1',
+}
+)
+uci:section('firewall', 'zone', 'mmfd',
+{
+	name = 'mmfd',
+	input = 'REJECT',
+	output = 'accept',
+	forward = 'REJECT',
+	device = 'mmfd+',
+	log = '1',
+}
+)
+
+-- forwardings and respective rules
+uci:section('firewall', 'forwarding', 'fcc',
+{
+	src = 'client',
+	dest = 'client',
+}
+)
+
+uci:section('firewall', 'forwarding', 'fcm',
+{
+	src = 'client',
+	dest = 'mesh',
+}
+)
+
+uci:section('firewall', 'forwarding', 'fmc',
+{
+	src = 'mesh',
+	dest = 'client',
+}
+)
+
+uci:section('firewall', 'forwarding', 'fmm',
+{
+	src = 'mesh',
+	dest = 'mesh',
+}
+)
+
+uci:section('firewall', 'forwarding', 'flc',
+{
+	src = 'l3roamd',
+	dest = 'client',
+}
+)
+
+uci:section('firewall', 'forwarding', 'fcl',
+{
+	src = 'client',
+	dest = 'l3roamd',
+}
+)
+
+--- rules for wan
+uci:section('firewall', 'rule', 'wan_fastd',
+{
+	name = 'wan_fastd',
+	dest = 'wan',
+	dest_port = '10000:10010',
+	proto = 'udp',
+	target = 'ACCEPT',
+}
+)
```

Comment:
this is about outgoing connections, other than that we probably cannot assume this.....
maybe the ports should be generated from site.conf

Good catch!